lms/djangoapps/email_marketing/tasks.py

```diff
@@ -0,0 +1,163 @@
+"""
+This file contains celery tasks for email marketing signal handler.
+"""
+import logging
+import time
+
+from pytz import UTC
+
+from celery import task
+from django.contrib.auth.models import User
+
+from student.models import UserProfile
+from email_marketing.models import EmailMarketingConfiguration
+
+from sailthru.sailthru_client import SailthruClient
+from sailthru.sailthru_error import SailthruClientError
+
+log = logging.getLogger(__name__)
+
+
+# pylint: disable=not-callable
+@task(bind=True, default_retry_delay=3600, max_retries=24)
+def update_user(self, username, new_user=False, activation=False):
+    """
+    Adds/updates Sailthru profile information for a user.
+     Args:
+        username(str): A string representation of user identifier
+    Returns:
+        None
+    """
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.enabled:
+        return
+
+    # get user
+    user = getUserAndProfile(username)
```

Comment:
Is this only split out so that you can patch it in your tests? If so, it might be better to just use a `UserFactory` to create the users you need, rather than patching this method.